app/controllers/api/base_controller.rb

```diff
@@ -21,14 +21,24 @@ def authenticate!
     end
 
     def current_provider
-      @current_provider ||= Provider.first
+      @current_provider ||= auth_token.provider
     end
 
   private
 
     def valid_authentication_token?
-      # TODO: Replace this with a proper authentication check
-      request.headers["Authorization"] == "Bearer bat"
+      auth_token.present? && auth_token.enabled?
+    end
+
+    def auth_token
+      @auth_token ||= begin
+        bearer_token = request.headers["Authorization"]
+        raise("Bearer token is blank") if bearer_token.blank?
```

Comment:
maybe just `return nil if bearer_token.blank?` so that you don't need the `rescue`?

Reply:
That's actually how I had it initially, but rubocop moaned at me for returning inside of a begin block... I've reworked it according to the example [here](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/NoReturnInBeginEndBlocks), let me know what you think.